images/kindnetd/cmd/kindnetd/subnets.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"os"
+	"regexp"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/utils/net"
+)
+
+const (
+	kubeadmClusterCIDRRegex   = `\s+podSubnet: (.*)\n`
+	kubeproxyClusterCIDRRegex = `\s+clusterCIDR: (.*)\n`
+)
+
+type kubeSubnets struct {
+	regex     string
+	configmap string
+	namespace string
+}
+
+func newKubeSubnets(regex, configmap, namespace string) *kubeSubnets {
+	return &kubeSubnets{
+		regex:     regex,
+		configmap: configmap,
+		namespace: namespace,
+	}
+}
+
+func (ks *kubeSubnets) Get(clientset *kubernetes.Clientset) (string, error) {
+	cidrRegexp := regexp.MustCompile(ks.regex)
```

Comment:
code from here to next comment should just be a function.
also it should have retries with backoff.